### ISSUE TITLE
Reactor Planner Designs Not Stable

### ISSUE THREAD

**Issue by MachiavelliSeraphim** [TinyModularThings/IC2Classic#381]
Some of the reactor designs that the planner claim are stable after multiple runs melt down when actually built.

Ive got a steam design that the planner shows rock solid that just cratered 3 times in a row in under 5 minutes run time.

Ill post the planner code and screenshots when I get back to my desk.

**Comment by Speiger** [comment]
Yeah would be nice.
The planner isnt 100% accurate thats right and it is kinda impossible to gain 100% accuracy since I have to simulate the reinsertion of the items etc. Though it shouldnt be to inaccurate.
Best 98%, worst 95% (i used it plenty of times)

**Comment by MachiavelliSeraphim** [comment]
I usually design with at least a good 20% margin above and beyond.  Im used to Reactorcraft and its forgiving nature "pun intended".  I messed with the plannner for hours and was stable every time.  Then i built it in a test world.  Then rebuilt the test area after the 1st 2nd and 3rd failures.  Each in under 90 seconds :(

Guess I just need to brush up on my IC2 skills... lol.  

I will still post everything when I get back.

**Comment by MachiavelliSeraphim** [comment]
4 Creative TE tanks delivering water.
4 Resonant Servos drawing out Steam to tanks.
Tanks buffering steam to turbine.
Turbine consuming 2000 mb/tick.

![Layout](https://user-images.githubusercontent.com/10451525/69486758-2244dc00-0e1d-11ea-9d9e-85c1d385fc37.png)
![Water Supply](https://user-images.githubusercontent.com/10451525/69486762-283abd00-0e1d-11ea-8bb5-c836b698e068.png)
![Steam Lines-Tank](https://user-images.githubusercontent.com/10451525/69486763-2c66da80-0e1d-11ea-9bb5-4226e6e2f3ad.png)

**Comment by MachiavelliSeraphim** [comment]
This was the live runtime.

STARTUP
![Startup +0 Sec](https://user-images.githubusercontent.com/10451525/69486775-3f79aa80-0e1d-11ea-8750-40700f43735b.png)
STARTUP +5
![Startup +5 Sec](https://user-images.githubusercontent.com/10451525/69486776-42749b00-0e1d-11ea-8e41-4ca7a75d8614.png)
STARTUP +6
![Startup +6 Sec](https://user-images.githubusercontent.com/10451525/69486779-46082200-0e1d-11ea-84d3-852191aaddfa.png)
STARTUP +7
![Startup +7 Sec](https://user-images.githubusercontent.com/10451525/69486780-486a7c00-0e1d-11ea-83db-c0f00cfc3c91.png)
STARTUP +8
![Startup +8 Sec](https://user-images.githubusercontent.com/10451525/69486783-4b656c80-0e1d-11ea-911b-546f41ea1435.png)
SHUTDOWN BEFORE CRATER AGAIN.

**Comment by MachiavelliSeraphim** [comment]
Reacror planner shows full cycle stable.
![Reactor Planner Complete Cycle](https://user-images.githubusercontent.com/10451525/69486810-bc0c8900-0e1d-11ea-82f1-cb97b9e5e5c3.png)

**Comment by Speiger** [comment]
you know this system produces 40 buckets of steam per second? That means the buffer just overflows?

**Comment by Speiger** [comment]
yeah after i fixed the fluid export of the steam that didnt happen.
TLDR: Steamvents stop cooling themselves when they cant export steam.
Your setup produces 40 buckets of steam (ruffly) per second. You need a way to export that somehow.
(Machine tanks + stacked fluid import/export upgrades can handle that but you need somewhere to get it out of the reactor and worst case void it)

Not a bug. (Tested live)

**Comment by Speiger** [comment]
@MachiavelliSeraphim ping

**Comment by MachiavelliSeraphim** [comment]
Sorry for the delay.  That turbine eats 2 buckets per tick, so 40/second.
Well 1920 mb/tick is what its throttled at.

I know the tank and laminar ducts can handle that flow because I use that setup on other steam sources with the same turbine.

Thats why I was puzzled why it overheated.
With the turbine on, the tanks stay steady.  Then it explodes.....

Id completely understand if they filled and caused a backup....

**Comment by Speiger** [comment]
@MachiavelliSeraphim I tested the design with the design with steam reactor. Yes it almost exploded but as soon as I fixed the transportation it didnt explode.
How i did it:
I had a machine tank next to the reactor, and had a stack of fluid import upgrades facing to the reactor and then had a creative upgrade in the machine tank too (it voids fluids for testing like this)

And the reactor stablized like it did in the planner. I had it running for around 30 seconds before finishing the test.

I think that your TE pipes can not transfer 2 Buckets per tick? So maybe your pipes are the issue. If thats the case then use machine buffers with i think 2-3 fluid import & export upgrades. (Not the basic ones)
Also these upgrades need EU to run. 1EU per 100mB transfered. So around 40-60EU/t (for import & export combined)

**Comment by MachiavelliSeraphim** [comment]
I know the pipes can handle the load, but maybe theres tick lag on the servos.  Ive moved some ridiculous quantities of fluid with those in reactorcraft and rotarycraft.

Ill try the tank buffer and see what happens.

Thanks.

P.s.  is there a good writeup on how the reactor tracker is used?
Id like to set one up for automatic refueling if i can.

Thanks again.

**Comment by Speiger** [comment]
Yeah that could be your problem because the reactor ticks heat 20 times a second, and the buffer isnt big enough for 1 second worth of heat. Maybe 0.5 seconds. Though with the machine buffer you could at least grow the capacity of the tanks that you have. Because they have 64 buckets of storage. So having 2 of them as buffers (128 buckets) and then drain from them might be your solution because you get at least 3-4 seconds of time before the tanks are full. And if you use fast import upgrades they will be able to keep up.

The reactor tracker is basically just a item exchanger. It is odly specific and kinda a nessesary evil do to the design of how the reactor components were designed. I shouldn't have done it that way and that was a massive fault.
So to explain what the track upgrade does:
It is basically a specialized import & export upgrade depending on what you set it to.
When you want to use the upgrade you need to initialize it first on a exsting reactor wtih all the items that you want inserted or be only present. It makes a template basically.
Before you do that though you need to decide which type of reactor component it should track.
(Based on the filters that you have in the reactor planner)
- In insert mode it will basically remove items from the machine buffer or any machine that can hold the upgrade and insert it into the reactor into the slots that you have in the template. Basically trying to keep replace items that are missing from the template.
- In Export mode it basically scans the reactor for any invalid item and tries to insert it into the machine buffer or the machine that it is in so that the reactor keeps the inventory with all valid items.

And thats all that is to it. Sadly i can not fix this poor design because that would void all the reactor components, and that is to world breaking.

These upgrades are only nessesary if you dont have a full reactor setup or a breeder so that it functions easier. But for a setup like yours a normal insertion & extraction pipe from TE should be plenty because you can not make a lot wrong by just exchanging used rods with new rods.

**Comment by MachiavelliSeraphim** [comment]
How do you change it from export to import?  Import sounds perfect for my breeder.

**Comment by MachiavelliSeraphim** [comment]
Apparently im dumb or something.  
I cant figure out how to link the tracker or switch it to import....  it doesnt seem to like anything I do.

**Comment by Speiger** [comment]
(Rightclick actions)
Side inventory key on reactor to scan it. (after you set the mode settings)
Modeswitch key to toggle insert/extract.
alt + mode key to change the scan type

**Comment by MachiavelliSeraphim** [comment]
Thanks.  So far this looks amazing.  Much better than the Gregtech Advanced Machine Buffer I used years ago.

**Comment by MachiavelliSeraphim** [comment]
Best contraption ever.  Its ingenious.
Bit of a hassle to configure, but all great things are. 
Thanks for all the help.
